packages/qwik/src/optimizer/core/src/transform.rs

```diff
@@ -838,27 +838,16 @@ impl<'a> QwikTransform<'a> {
 			get_null_arg()
 		};
 
-		let (jsx_func, mut args) = if is_fn {
+		// TODO use _restProps or similar to remove const props from spread props
+		let (jsx_func, mut args) = if is_fn || has_spread_props {
 			(
 				self.ensure_core_import(&_JSX_C),
-				vec![node_type, mutable_props, flags, key],
-			)
-		} else if dynamic_props {
-			(
-				self.ensure_core_import(&_JSX_S),
-				vec![node_type, mutable_props, immutable_props, flags, key],
+				vec![node_type, var_props, const_props, flags, key],
```

Comment:
potential difference

Reply:
Not sure what you mean, this will definitely give different results than before